leader/leader.go

```diff
@@ -105,15 +105,20 @@ func (l *Client) AddWatchCallback(key string, retry time.Duration, fn CallbackFn
 	}()
 }
 
-func (l *Client) getWatchAtLatestIndex(ctx context.Context, api client.KeysAPI, key string, retry time.Duration) (client.Watcher, *client.Response, error) {
-	resp, err := l.getFirstValue(key, retry)
+func (l *Client) recreateWatchAtLatestIndex(ctx context.Context, api client.KeysAPI, key string, retry time.Duration) (client.Watcher, *client.Response, error) {
+	prefix := ctx.Value("prefix")
+	log := log.WithFields(log.Fields{"watch": prefix})
+
+	resp, err := api.Get(ctx, key, nil)
 	if err != nil {
-		return nil, nil, trace.BadParameter("%v unexpected error: %v", ctx.Value("prefix"), err)
-	} else if resp == nil {
-		log.Debugf("%v client is closing, return", ctx.Value("prefix"))
+		return nil, nil, trace.Wrap(err, "failed to fetch value")
+	}
+	if resp == nil {
+		log.Debug("client is closing")
 		return nil, nil, nil
```

Comment:
I know this is not a part of your changes, but `nil, nil, nil` is a pretty bad return value as it can potentially lead to a panic if the caller expects first values to be usable in case of nil error. If we changed this to return an error, do you know if this would require a lot of refactoring in code that uses this?